### ISSUE TITLE
Add support for the VideoCapture::retrieve method. 

### ISSUE THREAD

**Issue by khushbuadav** [hybridgroup/gocv#928]
## Description
I’m trying to capture the latest frames from an RTSP stream and doing some processing which adds delay in fetching the next frame from the stream. 
I don’t need to read all the missed frames but get the latest frame and skip the remaining from the buffer.
I am planning to use two goroutines. One go routine will continuously grab(VideoCapture::grab) the frames and not process them. 
The second go routine will be used to retrieve the latest frame and process it. 

Right now I am using VideoCapture::read in the second go routine which internally calls VideoCapture::grab. And because of this, I am getting below error during read. 
```
Assertion fctx->async_lock failed at src/libavcodec/pthread_frame.c:155
SIGABRT: abort
PC=0x7fdcb53797bb m=0 sigcode=18446744073709551610
```
I assumed this is because both the go routine are calling grab at the same time. 

Ideally, in the second go routine we should just call [retrieve](https://docs.opencv.org/3.4/d8/dfe/classcv_1_1VideoCapture.html#a9ac7f4b1cdfe624663478568486e6712) and get the latest from the buffer. However, I did not find the retrieve method in gocv. 

## Your Environment
* GoCV version used: 0.29.0
* Go version: 1.17.2 darwin/amd64

**Comment by quanqigu** [comment]
is this implemented? @deadprogram

**Comment by deadprogram** [comment]
`retrieve()` is actually not implemented, looks like it was overlooked.

A PR to add this would be welcomed!

Please see https://github.com/hybridgroup/gocv/blob/release/CONTRIBUTING.md#something-you-wantneed-from-opencv-does-not-appear-to-be-in-gocv

**Comment by deadprogram** [comment]
Released as part of 0.36 so now closing. Thank you!
